contracts/SnappAuction.sol

```diff
@@ -22,6 +36,15 @@ contract SnappAuction is SnappBase {
         uint128 sellAmount
     );
 
+    event StandingSellOrderBatch(
+        uint validFromAuctionIndex, 
```

Comment:
Thinking about this again, do we then even need a nonce? If validFromAuctionIndex uniquely identifies the batch what's the point of nonce?

Reply:
I guess once we allow rollbacks validFromAuction could become smaller than a previous value. in that case, I believe we should emit nonce as part of the event and not the auction index.

Reply:
The smart contract needs to know the previous ordersubmission, in order to set the validToAuctionIndex.

But of course, this data can be submitted as payload to the function. After reading your recommended book, I did not implement it, in order to keep the code simpler. Maybe I should, as this would save gas costs.

Yeah, I am not sure, how relevant the rollbacks are. I would suggest that traders are supposed to verify the state and do not submit orders if the state is illegal. If they do, then they are committed to those orders, even when auctions are reversed.

What do you guys think, should I implement that the previous nonce is handed over via payload and verified by checking that the validToAuctionIndex of this orderBatch is not set?

Reply:
Hm, let me do a quick PR into this PR.
Having the code is easier to discuss.

Reply:
I created the PR

Reply:
I don't know what is happening here anymore.